stream/ResponseCollector.java

```diff
@@ -64,7 +65,8 @@ public synchronized void close(@Nullable StatusRuntimeException error) {
         private final BlockingQueue<Either<Response<R>, Done>> responseQueue;
 
         Queue() {
-            responseQueue = new LinkedBlockingQueue<>();
+            // TODO: We should be using LinkedBlockingQueue here, however there is a bug in Java 14-16: see #351
```

Comment:
Should we make it a bit more concise:

```
// TODO: switch LinkedTransferQueue to LinkedBlockingQueue once issue #351 is fixed
```

Reply:
I'm actually considering getting rid of the "TODO" entirely @lolski . Even if _we_ upgrade to Java 18 in the future, there's no guarantee that our users will. Therefore I think it's safer to keep `LinkedTransferQueue` and allow ourselves to support Java 14-16 for the foreseeable future (at least 2 years)

Reply:
I think we can expect the bug to be fixed on a newer Java 14-16. You know how Java 8 gets tons of minor updates? I would expect the same for Java 14-16.

Reply:
Alright, changed